experimental/tools/unit-generators/vhdl/generators/handshake/load.py

```diff
@@ -121,21 +123,22 @@ def _generate_load_signal_manager(name, data_bitwidth, addr_bitwidth, extra_sign
       "extra_signals": extra_signals
   }])
 
-  # Only extra signals (not data) are concatenated, so set inner port bitwidth to 0.
-  addrIn_inner_port = {
-      "name": "addrIn",
-      "bitwidth": 0,
-      "extra_signals": extra_signals
+  # Only extra signals (not data) are concatenated, so set original port bitwidth to 0.
+  addrIn_conversion: ConcatPortConversion = {
+      "original_name": "addrIn",
+      "original_bitwidth": 0,
+      "inner_name": "addrIn_inner"
   }
-  dataOut_inner_port = {
-      "name": "dataOut",
-      "bitwidth": 0,
-      "extra_signals": extra_signals
+  dataOut_conversion: ConcatPortConversion = {
+      "original_name": "dataOut",
+      "original_bitwidth": 0,
+      "inner_name": "dataOut_inner"
   }
   concat_signal_decls = generate_concat_signal_decls(
-      [addrIn_inner_port, dataOut_inner_port], extra_signals_total_bitwidth)
-  concat_signal_logic = generate_concat_logic(
-      [addrIn_inner_port], [dataOut_inner_port], concat_info)
+      [addrIn_conversion, dataOut_conversion], extra_signals_total_bitwidth)
+
+  concat_signal_logic = generate_concat_port_assignments(
+      [addrIn_conversion], [dataOut_conversion], concat_info)
```

Comment:
`generate_concat_signal_decls` and `generate_concat_logic` (renamed to `generate_concat_port_assignments`) now take `list[ConcatPortConversion]` instead of `list[Port]`.

This is because:

- These methods don't use the `extra_signals` field in `Port`, so passing full `Port` objects was unnecessary and confusing (as previously suggested in another context by @gioelegott).
- The `inner_name` (name of the concatenated signal) used to be set inside these methods, but this responsibility now properly belongs to the signal manager itself.